lib/msgpax.ex

```diff
@@ -1,17 +1,131 @@
 defmodule Msgpax do
+  @moduledoc """
+  This module provides functions for serializing and de-serializing Elixir terms
+  using the [MessagePack](http://msgpack.org/) format.
+  """
+
   alias __MODULE__.Packer
   alias __MODULE__.Unpacker
 
-  defdelegate [pack(term), pack!(term)], to: Packer
+  @doc """
+  Serializes `term`.
+
+  This function returns iodata; if you want to force the result to be a binary,
+  you can use `IO.iodata_to_binary/1`.
+
+  This function returns `{:ok, iodata}` if the serialization is sucessful,
+  `{:error, reason}` otherwise. Reason can be:
+
+    * `{:bad_arg, term}` - means that the given argument is not serializable. For
+      example, this is returned when you try to pack a bitstring (as only
```

Comment:
I think we can use "bits" instead of "bitstring" to make differentiation from "binaries" even better.